doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -220,63 +220,79 @@ Requests to the environment
+.. cmd:: Print Opaque Dependencies @smart_qualid
 
-      Displays the set of transparent constants :n:`@qualid` relies on
-      in addition to the assumptions.
+   Displays the assumptions and opaque constants that :n:`@smart_qualid` depends on.
 
-   .. cmdv:: Print All Dependencies @qualid
-      :name: Print All Dependencies
+.. cmd:: Print Transparent Dependencies @smart_qualid
 
-      Displays all assumptions and constants :n:`@qualid` relies on.
+   Displays the assumptions and  transparent constants that :n:`@smart_qualid` depends on.
 
+.. cmd:: Print All Dependencies @smart_qualid
 
-.. cmd:: Search @qualid
+   Displays all the assumptions and constants :n:`@smart_qualid` relies on.
 
-   This command displays the name and type of all objects (hypothesis of
-   the current goal, theorems, axioms, etc) of the current context whose
-   statement contains :n:`@qualid`. This command is useful to remind the user
-   of the name of library lemmas.
 
-   .. exn:: The reference @qualid was not found in the current environment.
+.. cmd:: Search {+ @search_item } {? {| inside | outside } {+ @qualid } }
 
-      There is no constant in the environment named qualid.
+   .. insertprodn search_item search_item
+
+   .. prodn::
+      search_item ::= {? - } @string {? % @ident }
+      | {? - } @one_term
+
+   This command displays the name and type of objects (hypothesis of
+   the current goal, theorems, axioms, etc) in the current context.
+   This command is useful to remind the user of the names of library lemmas.
 
-   .. cmdv:: Search @string
+   .. todo above: "current context" meaning on the current goal or the first goal?
+      below: adapted the old wording, doesn't seem consistent
+      also will "invocation-of-tactics" cover the command case, or will there be a new section?
 
-      If :n:`@string` is a valid identifier, this command
-      displays the name and type of all objects (theorems, axioms, etc) of
-      the current context whose name contains string. If string is a
-      notation’s string denoting some reference :n:`@qualid` (referred to by its
-      main symbol as in `"+"` or by its notation’s string as in `"_ + _"` or
-      `"_ 'U' _"`, see Section :ref:`notations`), the command works like ``Search`` :n:`@qualid`.
+   * :n:`@string  {? % @ident }` - If :n:`@string` is a valid identifier,
+     search for objects whose name contains :n:`@string`. If :n:`@string` is a
+     notation's string for some reference :n:`@qualid` (referred to by its
+     main symbol as in `"+"` or by its notation’s string as in `"_ + _"` or
+     `"_ 'U' _"`, see Section :ref:`notations`) the command works like :cmd:`Search` :n:`@qualid`.
 
-   .. cmdv:: Search @string%@ident
+     .. todo: don't understand the 2nd sentence; notation's string -> notation string?
+        what if it's not a valid identifier?  error or does something different?
+        this is a substring match, e.g. "abc" matches "bc", right?
```

Comment:
The terminology "valid identifier" is misleading. What is meant is valid part of an identifier (which grammatically speaking is almost the same except that keywords and `_` work too). Example: `Search "in".`